src/geometry/regionbasedadjacency.h

```diff
@@ -34,27 +34,35 @@ class RegionBasedAdjacency
 {
 public:
     RegionBasedAdjacency();
+    RegionBasedAdjacency(const RegionBasedAdjacency &other);
+    RegionBasedAdjacency &operator=(const RegionBasedAdjacency &other);
+    virtual ~RegionBasedAdjacency() {}
+
+#ifdef LIBGEODECOMP_WITH_CPP14
+    RegionBasedAdjacency(RegionBasedAdjacency &&other) = default;
+    RegionBasedAdjacency &operator=(RegionBasedAdjacency &&other) = default;
+#endif // LIBGEODECOMP_WITH_CPP14
 
     /**
      * Insert a single edge (from, to) to the graph
      */
-    void insert(int from, int to);
+    virtual void insert(int from, int to);
```

Comment:
I don't like the idea of the user having to subclass from Adjacency as it requires users to know how our code is working internally (see mail for more details). Please remove these virtual qualifiers.